cai_causal_graph/time_series_causal_graph.py

```diff
@@ -1007,6 +1007,50 @@ def to_numpy_by_lag(self) -> Tuple[Dict[int, numpy.ndarray], List[str]]:
         assert self.variables is not None
         return adjacency_matrices, self.variables
 
+    def get_nodes_at_lag(self, time_lag: int = 0) -> List[TimeSeriesNode]:
+        """Return all nodes at time delta `time_lag`."""
+        return [node for node in self.get_nodes() if node.time_lag == time_lag]
+
+    def get_contemporaneous_nodes(self, node: NodeLike) -> List[TimeSeriesNode]:
+        """Return all nodes that are contemporaneous to the provided node."""
+        assert node is not None, 'The `node` cannot be None.'
+        if isinstance(node, str):
+            node = self.get_node(node)
+
+        contemporaneous_nodes = []
+        assert isinstance(node, TimeSeriesNode), 'The node must be a `TimeSeriesNode`.'
+
+        for other_node in self.get_nodes():
+            assert isinstance(other_node, TimeSeriesNode)
+            if other_node.identifier != node.identifier and other_node.time_lag == node.time_lag:
+                contemporaneous_nodes.append(node)
+        return contemporaneous_nodes
+
+    def get_contemporaneous_adj_nodes(self, node: NodeLike) -> List[TimeSeriesNode]:
+        """Return the adjacent that are contemporanous to the provided node."""
```

Comment:
Return the nodes adjacent to the provided node that are also contamperaneous (i.e. have the same time_lag).

Reply:
Also - I would assume "adjacent" nodes are any nodes which are children or parents of this node, but here you seem to be only getting nodes which are parents (i.e. directed edge from them to this node) or potential parents (i.e. undirected edge between them and this node). What about children? Is this intentional? If so - I believe it should be documented in the doc string and potentially would change the name?

Or maybe I just misunderstood the code - in which case my apologies :)

Reply:
I think having a more thorough description of what this method returns would be beneficial

Reply:
I just want the true and potential parents. I will make it clear in the description.

Reply:
Also modified the method name to `get_potential_parents`